src/mouse.c

```diff
@@ -202,101 +202,105 @@ void clickMouse(MMMouseButton button)
  */
 void doubleClick(MMMouseButton button)
 {
-	
+
 #if defined(IS_MACOSX)
 
 	/* Double click for Mac. */
 	const CGPoint currentPos = CGPointFromMMPoint(getMousePos());
 	const CGEventType mouseTypeDown = MMMouseToCGEventType(true, button);
 	const CGEventType mouseTypeUP = MMMouseToCGEventType(false, button);
 
-	CGEventRef event = CGEventCreateMouseEvent(NULL, mouseTypeDown, currentPos, kCGMouseButtonLeft);  
-	
-	/* Set event to double click. */						
+	CGEventRef event = CGEventCreateMouseEvent(NULL, mouseTypeDown, currentPos, kCGMouseButtonLeft);
+
+	/* Set event to double click. */
 	CGEventSetIntegerValueField(event, kCGMouseEventClickState, 2);
-											
-	CGEventPost(kCGHIDEventTap, event);  
-																
-	CGEventSetType(event, mouseTypeUP);  
-	CGEventPost(kCGHIDEventTap, event);  
 
-	CFRelease(event); 
+	CGEventPost(kCGHIDEventTap, event);
+
+	CGEventSetType(event, mouseTypeUP);
+	CGEventPost(kCGHIDEventTap, event);
+
+	CFRelease(event);
 
 #else
 
 	/* Double click for everything else. */
 	clickMouse(button);
 	microsleep(200);
 	clickMouse(button);
-	
+
 #endif
 }
 
-/**
- * Function used to scroll the screen in the required direction.
- * This uses the magnitude to scroll the required amount in the direction. 
- * TODO Requires further fine tuning based on the requirements.
- */
-void scrollMouse(int scrollMagnitude, MMMouseWheelDirection scrollDirection)
+void scrollMouse(int x, int y)
 {
-	#if defined(IS_WINDOWS)
-		// Fix for #97 https://github.com/octalmage/robotjs/issues/97,
-		// C89 needs variables declared on top of functions (mouseScrollInput)
-		INPUT mouseScrollInput;
-	#endif
-
-	/* Direction should only be considered based on the scrollDirection. This
-	 * Should not interfere. */
-	int cleanScrollMagnitude = abs(scrollMagnitude);
-	if (!(scrollDirection == DIRECTION_UP || scrollDirection == DIRECTION_DOWN))
-	{
-		return;
-	}
-
-	/* Set up the OS specific solution */
-	#if defined(__APPLE__)
+#if defined(IS_WINDOWS)
+	// Fix for #97 https://github.com/octalmage/robotjs/issues/97,
+	// C89 needs variables declared on top of functions (mouseScrollInput)
+	INPUT mouseScrollInputH;
+	INPUT mouseScrollInputV;
+#endif
 
-		CGWheelCount wheel = 1;
-		CGEventRef event;
+  /* Direction should only be considered based on the scrollDirection. This
+   * Should not interfere. */
 
-		/* Make scroll magnitude negative if we're scrolling down. */
-		cleanScrollMagnitude = cleanScrollMagnitude * scrollDirection;
+  /* Set up the OS specific solution */
+#if defined(__APPLE__)
 
-		event = CGEventCreateScrollWheelEvent(NULL, kCGScrollEventUnitLine, wheel, cleanScrollMagnitude, 0);
-		CGEventPost(kCGHIDEventTap, event);
+	CGEventRef event;
 
-	#elif defined(USE_X11)
+	event = CGEventCreateScrollWheelEvent(NULL, kCGScrollEventUnitPixel, 2, y, x);
+	CGEventPost(kCGHIDEventTap, event);
 
-		int x;
-		int dir = 4; /* Button 4 is up, 5 is down. */
-		Display *display = XGetMainDisplay();
+  CFRelease(event);
```

Comment:
Sorry I meant that there are a few places where spaces are used instead of tabs. If you select the characters before CFRelease release you'll see what I mean. I can fix them after this gets merged though, if you'd like!